storage/range.go

```diff
@@ -1557,14 +1565,12 @@ func (r *Range) InitialState() (raftpb.HardState, raftpb.ConfState, error) {
 	return hs, cs, nil
 }
 
-// loadFirstAndLastIndex looks in the engine to find the first and last log index.
-// TODO(bdarnell): remove unnecessary calls to FirstIndex in etcd/raft and then load
-// the first index lazily (so we don't need to scan the whole range at startup)
-func (r *Range) loadFirstAndLastIndex() error {
+// loadLastIndex looks in the engine to find the last log index.
+func (r *Range) loadLastIndex() error {
 	logKey := engine.RaftLogPrefix(r.Desc().RaftID)
 	kvs, err := engine.MVCCScan(r.rm.Engine(),
```

Comment:
Wouldn't hurt to just throw in a comment saying that the raft log is indexed in reverse order, from most recent to least.